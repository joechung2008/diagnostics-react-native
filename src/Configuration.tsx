import React, { useMemo } from 'react';
import { Text, View } from 'react-native';
import { useTheme } from './ThemeContext';

const Configuration: React.FC<ConfigurationProps> = ({ config }) => {
  const { colors } = useTheme();

  const items = useMemo(
    () =>
      Object.entries(config).reduce<KeyValuePair<string>[]>(
        (previous, [key, value]) => [...previous, { key, value }],
        [],
      ),
    [config],
  );

  const dynamicStyles = {
    container: {
      backgroundColor: colors.background,
    },
    title: {
      fontSize: 16,
      fontWeight: 'bold' as const,
      marginBottom: 10,
      color: colors.text,
    },
    table: {
      borderWidth: 1,
      borderColor: colors.tableBorder,
      backgroundColor: colors.surface,
    },
    header: {
      flexDirection: 'row' as const,
      borderBottomWidth: 1,
      borderBottomColor: colors.tableBorder,
      paddingTop: 5,
      paddingBottom: 5,
      backgroundColor: colors.tableHeaderBg,
    },
    headerText: {
      flex: 1,
      fontWeight: 'bold' as const,
      paddingLeft: 5,
      paddingRight: 5,
      color: colors.text,
    },
    row: {
      flexDirection: 'row' as const,
      paddingTop: 5,
      paddingBottom: 5,
      borderBottomWidth: 1,
      borderBottomColor: colors.tableBorder,
    },
    cell: {
      flex: 1,
      paddingLeft: 5,
      paddingRight: 5,
      minWidth: 0, // Allow flex items to shrink below their content size
    },
    cellText: {
      flex: 1,
      flexShrink: 1,
      color: colors.text,
    },
  };

  return (
    <View style={dynamicStyles.container}>
      <Text style={dynamicStyles.title}>Configuration</Text>
      <View style={dynamicStyles.table}>
        <View style={dynamicStyles.header}>
          <Text style={dynamicStyles.headerText}>Key</Text>
          <Text style={dynamicStyles.headerText}>Value</Text>
        </View>
        {items.map(item => (
          <View key={item.key} style={dynamicStyles.row}>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.key}
            </Text>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.value}
            </Text>
          </View>
        ))}
      </View>
    </View>
  );
};

export default Configuration;
