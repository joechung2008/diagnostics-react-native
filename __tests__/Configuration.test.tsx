import Configuration from '@/Configuration';
import { render, screen } from '@testing-library/react-native';
import React from 'react';

// Mock the ThemeContext
jest.mock('@/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f8f9fa',
      text: '#212529',
      tableBorder: '#dee2e6',
      tableHeaderBg: '#f8f9fa',
    },
  }),
}));

describe('Configuration', () => {
  it('renders correctly with configuration data', () => {
    const config = {
      apiUrl: 'https://api.example.com',
      timeout: '30000',
      debug: 'true',
    };
    render(<Configuration config={config} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with empty configuration', () => {
    const config = {};
    render(<Configuration config={config} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
