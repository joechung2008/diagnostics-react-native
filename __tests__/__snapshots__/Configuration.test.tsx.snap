// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Configuration renders correctly with configuration data 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
    }
  }
>
  <Text
    style={
      {
        "color": "#212529",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Configuration
  </Text>
  <View
    style={
      {
        "backgroundColor": "#f8f9fa",
        "borderColor": "#dee2e6",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#f8f9fa",
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#212529",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#212529",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        apiUrl
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        https://api.example.com
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        timeout
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        30000
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        debug
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        true
      </Text>
    </View>
  </View>
</View>
`;

exports[`Configuration renders correctly with empty configuration 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
    }
  }
>
  <Text
    style={
      {
        "color": "#212529",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Configuration
  </Text>
  <View
    style={
      {
        "backgroundColor": "#f8f9fa",
        "borderColor": "#dee2e6",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#f8f9fa",
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#212529",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#212529",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
  </View>
</View>
`;
